include/swift/AST/DiagnosticsSema.def

```diff
@@ -2758,6 +2758,16 @@ ERROR(unexportable_clang_function_type,none,
       "it may use anonymous types or types defined outside of a module",
       (Type))
 
+ERROR(cxx_class_instantiation_failed,none,
+      "clang couldn't instantiate class template",
+      ())
+ERROR(cxx_class_instantiation_non_tag_argument,none,
+      "only struct/class/enum/union type arguments supported",
+      ())
+ERROR(cxx_class_instantiation_non_cxx_argument,none,
+      "only C++ types supported",
```

Comment:
Could you add tests for these new diagnostics?

Reply:
Also, for now, diagnostics need to be copied to localization/diagnostics/en.yaml as well. https://github.com/apple/swift/pull/32984#discussion_r456848980

Reply:
Two of these diagnostics were already tested in`test/Interop/Cxx/templates/class-template-instantiation-error-handling.swift`. The third one I couldn't come up with a test, so I converted the diagnostic with an assertion.